### PR TITLE
Defining structure and main functions

### PR DESCRIPTION
**Pull request by jonmchan**

### CODE REVIEW

File: app/views/pseudo_kiosk/authentication/unlock.html.erb [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+</head>
+
+<body translate="no">
+<div id="explanation_window" class="center-xy">
+    <h1>Thank you for your input!</h1>
+    <h2>Please pass the device back to the operator</h2>
+    <h1>Slide to Unlock</h1>
+    <input type="range" value="0" class="pullee" />
+</div>
+<% # This is bad... I am having a problem using the named routes within the rails engine plugin %>
```
Comment (BaneOfSerenity) [inline app/views/pseudo_kiosk/authentication/unlock.html.erb]: Still having trouble here?

File: lib/pseudo_kiosk/controller.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      #Config.configure!
+    end
+
+    module InstanceMethods
+      # To be used in before_action in the application_controller.
+      # If pseudo_kiosk enabled, all endpoints that are not in the kiosk_whitelist 
+      # will not allowed to be accessed
+      def secure_pseudo_kiosk
+        #  this needs to go to the unlock screen
+        # binding.pry if request.path_info == test_work_flow_complete_step3_privilege_path
```
Comment (BaneOfSerenity) [inline lib/pseudo_kiosk/controller.rb]: this can probably be removed now :D

File: app/views/pseudo_kiosk/authentication/unlock.html.erb (hunk 1 above)
Comment (jonmchan) [reply app/views/pseudo_kiosk/authentication/unlock.html.erb]: Yes, this is still an issue until we are able to resolve the engine loading issue we are having.

File: lib/pseudo_kiosk/controller.rb (hunk 2 above)
Comment (jonmchan) [reply lib/pseudo_kiosk/controller.rb]: I referenced this in a issue on rspec-rails, i'll leave it in there for them and whoever else needs to debug this. :)

**Pull request merged**